lib/health_cards/qbp_client.rb

```diff
@@ -0,0 +1,209 @@
+        phone_home[6] = patient_info[:phone][:local_number] # local number
+        qpd.phone_home = phone_home.join(msg_input.item_delim)
+      end
+
+      msg_input
+    end
+
+    # NOTE: This is a helper method to upload a patient to the IIS Sandbox and should be used for testing purposes only
+    # Send a VXU message to the IIS Sandbox service to upload a patient
+    # @param vxu_path [String] File Path where HL7 V2 VXU message is located
+    def upload_patient(vxu_path = 'lib/assets/vxu_fixtures/vxu.hl7')
+      # Define client
+      service_def = 'lib/assets/service.wsdl'
+      client = Savon.client(wsdl: service_def,
+                            endpoint: Rails.application.config.iisSandboxHost + '/iis-sandbox/soap',
+                            pretty_print_xml: true)
+      # Upload Patient from fixture
+      upload_raw_input = FILE.open(vxu_path).readlines
+      upload_msg_input = HL7::Message.new(upload_raw_input)
+      client.call(:submit_single_message) do
+        message({ username: Rails.application.config.username,
+                  password: Rails.application.config.password,
+                  facilityID: Rails.application.config.facilityID,
+                  hl7Message: upload_msg_input })
+      end
+    end
+
+    # Methods That Parse HL7 V2 Message
+
+    # Get the status of a response returned from the IIS Sandbox. Throw Errors for invalid responses.
+    # @param msg_response [HL7::Message] The response message returned from an IIS Sandbox QBP Query
+    # @return [Symbol] The status of the response from the IIS-Sandbox
+    # Return Options:
+    #   :OK = Data found, no errors (this is the default)
+    #   :NF = No data found, no errors
+    #   :AE = Application Error
+    #   :TM = Too much data found
+    def get_response_status(msg_response)
+      # Get QAK.2 (Query Response Status)
+      response_status = msg_response[:QAK][2].to_sym
+
+      # Handle Errors for each profile that could be returned; modify response_status if necessary
+      profile = msg_response[:MSH][20][0..2].to_sym
+      case profile
+      when :Z32
+        handle_z32_errors(msg_response)
+      when :Z31
+        handle_z31_errors(msg_response)
+        # Setting response status to :TM (Too much data found) to handle case where multiple mathces are returned.
+        # Query Response Status (QAK) segment would not indicate the need to input more information in this scenario.
+        response_status = :TM
+      when :Z33
+        handle_z33_errors(msg_response)
+      else
+        # TODO: Come up with request that can produce an unrecognized response profile
+        # (this may require a flavor in the IIS Sandbox)
+        raise HealthCards::OperationNotSupportedError
+      end
+
+      response_status
+    end
+
+    # Methods that check for and handle errors
+
+    def check_client_connectivity(client)
+      response = client.call(:connectivity_test) do
```

Comment:
This may throw a `Savon::Error` if it fails badly, so maybe wrap this in a `begin ... rescue` and then throw a QBPClient Error.

[Savon Error Doc](https://www.savonrb.com/version2/errors.html)

Reply:
Good catch. I has added this to the `:submit_single_message` call but forgot to for the connectivity test. I'll add error handling for Savon::Errors